xtokens/src/mock/mod.rs

```diff
@@ -58,13 +61,15 @@ impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
 				X2(Parachain(1), GeneralKey(k)) if k == a1 => Some(CurrencyId::A1),
 				X2(Parachain(2), GeneralKey(k)) if k == b => Some(CurrencyId::B),
 				X2(Parachain(2), GeneralKey(k)) if k == b1 => Some(CurrencyId::B1),
+				X1(Parachain(3)) => Some(CurrencyId::C),
 				_ => None,
 			},
 			MultiLocation { parents, interior } if parents == 0 => match interior {
 				X1(GeneralKey(k)) if k == a => Some(CurrencyId::A),
 				X1(GeneralKey(k)) if k == b => Some(CurrencyId::B),
 				X1(GeneralKey(k)) if k == a1 => Some(CurrencyId::A1),
 				X1(GeneralKey(k)) if k == b1 => Some(CurrencyId::B1),
+				Here => Some(CurrencyId::C),
```

Comment:
Parachains in mock shares the same `CurrencyIdConvert`, this will make all of them convert `Here` to `CurrencyId::C`, which doesn't make sense.

You'll need to define a convert impl only for Parachain `C`.